src/schemas.ts

```diff
@@ -38,6 +38,9 @@ const contentBlock = () =>
     level: z.number().optional(),
   });
 
+const slug = (fieldName: string) =>
+  [`${fieldName}.current`, z.string()] as [string, z.ZodType];
```

Comment:
```suggestion
  [`${fieldName}.current`, z.string()] as [string, z.ZodString];
```

With `[string, z.ZodType]`, the resulting data type for a slug field is `any` (since `z.ZodType` is broad). If we just tighten this down a tiny bit, we'll get `slug` fields typed as `string` instead (which I think is what we want).